### PR TITLE
Add C# API for CopyVariableValueTo for sparse data

### PR DESCRIPTION
**Pull request by liqunfu**
Add C# API for CopyVariableValueTo for sparse data and test

### CODE REVIEW

**Comment by zhouwangzw** [comment]
> // bool/double/float are already enabled with SWIG_STD_VECTOR_ENHANCED in std_vector.i

add "int/bool/...", since you have added vector<int> below.

---
Refers to: bindings/common/CNTKManagedCommon.i:43 in e87e0b9. [](commit_id = e87e0b91630fd3f3abdfae4f0d90fe1554c65413, deletion_comment = False)

File: bindings/csharp/Swig/cntk_cs.i [hunk 1]
```diff
@@ -1274,6 +1274,55 @@
         return;
     }
 
+	public void CopyVariableValueTo<T>( Variable outputVariable,
```
Comment (zhouwangzw) [inline bindings/csharp/Swig/cntk_cs.i]: Could you please change the method name to GetSparseData<T>(....), to align with those for dense and onehot. The CopyVariableValueTo for desne and onehot was deprecated now.

File: bindings/csharp/Swig/cntk_cs.i [hunk 2]
```diff
@@ -1274,6 +1274,55 @@
         return;
     }
 
+	public void CopyVariableValueTo<T>( Variable outputVariable, 
+										out int sequenceLength,
```
Comment (liqunfu) [inline bindings/csharp/Swig/cntk_cs.i]: replaced tab with spaces.

File: Examples/Evaluation/CNTKLibraryCSEvalCPUOnlyExamples/CNTKLibraryCSEvalExamples.cs [hunk 3]
```diff
@@ -738,6 +738,68 @@ public static void EvaluationSingleSequenceUsingSparse(DeviceDescriptor device)
             }
         }
 
+        public static void ValueCopyToSparseCSCTest<T>(DeviceDescriptor device)
```
Comment (liqunfu) [inline Examples/Evaluation/CNTKLibraryCSEvalCPUOnlyExamples/CNTKLibraryCSEvalExamples.cs]: Moved test code to memorytests.cs. Good to know the mechanism for packaging samples. I did not find models with sparse output. Will look into it and add samples once find one.

File: bindings/csharp/Swig/cntk_cs.i (hunk 1 above)
Comment (liqunfu) [inline bindings/csharp/Swig/cntk_cs.i]: Changed the function name to GetSparseData.

File: Tests/EndToEndTests/EvalClientTests/CNTKLibraryCSEvalExamplesTest/MemoryTests.cs [hunk 4]
```diff
@@ -60,5 +62,67 @@ public static void WriteOutputs()
             }
             Console.WriteLine("\nAll saved object references are printed.");
         }
+
```
Comment (zhouwangzw) [inline Tests/EndToEndTests/EvalClientTests/CNTKLibraryCSEvalExamplesTest/MemoryTests.cs]: Please add a comment that this test will be moved to the C# unit test project when it is ready.

File: bindings/common/CNTKValueExtend.i [hunk 5]
```diff
@@ -149,4 +149,22 @@
...
+    }
+
+    void CNTK::Value::CopyVariableValueToDouble(const Variable& outputVariable, int* sequenceLength, std::vector<SparseIndexType>& colStarts, 
+		std::vector<SparseIndexType>& rowIndices, std::vector<double>& nonZeroValues, int* numNonZeroValues)
+    {
+		size_t sequenceLengthSizeT, numNonZeroValuesSizeT;
+        self->CopyVariableValueTo<double>(outputVariable, sequenceLengthSizeT, colStarts, rowIndices, nonZeroValues, numNonZeroValuesSizeT);
+		*sequenceLength = (int)sequenceLengthSizeT;
+		*numNonZeroValues = (int)numNonZeroValuesSizeT;
+	}
```
Comment (zhouwangzw) [inline bindings/common/CNTKValueExtend.i]: I still see some hard tabs in lines above (L154-L168). Please replace it with spaces.

**Pull request closed without merging**